src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -113,5 +153,209 @@ private void InitializeScope(IStep step)
             var scopeName = step.ExecutionContext.ScopeName;
             step.ExecutionContext.ExpressionValues["steps"] = stepsContext.GetScope(scopeName);
         }
+
+        private void SetAllCompositeOutputs(
+            IExecutionContext parentExecutionContext,
+            DictionaryContextData actionOutputs)
+        {
+            // Each pair is structured like this
+            // We ignore "description" for now
+            // {
+            //   "the-output-name": {
+            //     "description": "",
+            //     "value": "the value"
+            //   },
+            //   ...
+            // }
+            foreach (var pair in actionOutputs)
+            {
+                var outputsName = pair.Key;
+                var outputsAttributes = pair.Value as DictionaryContextData;
+                outputsAttributes.TryGetValue("value", out var val);
+                var outputsValue = val as StringContextData;
+
+                // Set output in the whole composite scope. 
+                if (!String.IsNullOrEmpty(outputsName) && !String.IsNullOrEmpty(outputsValue))
+                {
+                    parentExecutionContext.SetOutput(outputsName, outputsValue, out _);
+                }
+            }
```

Comment:
just inline these into `HandleOutput` since the `SetAllCompositeOutputs()` only get used once.